src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -911,14 +917,20 @@ public boolean supportsOpenStatementsAcrossRollback() throws SQLException {
   public int getMaxBinaryLiteralLength() throws SQLException {
     logger.trace("int getMaxBinaryLiteralLength()", false);
     raiseSQLExceptionIfConnectionIsClosed();
-    return 8388608;
+    return getMaxCharLiteralLength() / 2; // hex instead of octal, thus divided by 2
   }
 
   @Override
   public int getMaxCharLiteralLength() throws SQLException {
     logger.trace("int getMaxCharLiteralLength()", false);
     raiseSQLExceptionIfConnectionIsClosed();
-    return 16777216;
+    int length = DEFAULT_MAX_LOB_SIZE;
+    Integer maxLiteralLengthFromSession =
+        (Integer) session.getOtherParameter(MAX_VARCHAR_BINARY_SIZE_PARAM_NAME);
+    if (maxLiteralLengthFromSession != null) {
+      length = maxLiteralLengthFromSession;
```

Comment:
What do you think about using Optional().orElse()

Reply:
Good idea, changed